mne/source_tfr.py

```diff
@@ -0,0 +1,305 @@
+            if sens_data.ndim != self._data_ndim:
+                raise ValueError('The sensor data must have %s dimensions, got'
+                                 ' %s' % (self._data_ndim, sens_data.ndim,))
+
+        if isinstance(vertices, list):
+            vertices = [np.asarray(v, int) for v in vertices]
+            if any(np.any(np.diff(v.astype(int)) <= 0) for v in vertices):
+                raise ValueError('Vertices must be ordered in increasing '
+                                 'order.')
+
+            n_src = sum([len(v) for v in vertices])
+
+            if len(vertices) == 1:
+                vertices = vertices[0]
+        elif isinstance(vertices, np.ndarray):
+            n_src = len(vertices)
+
+        # safeguard the user against doing something silly
+        if data is not None:
+            if data.shape[0] != n_src:
+                raise ValueError('Number of vertices (%i) and stfr.shape[0] '
+                                 '(%i) must match' % (n_src, data.shape[0]))
+            if data.ndim == self._data_ndim - 1:  # allow upbroadcasting
+                data = data[..., np.newaxis]
+            if data.ndim != self._data_ndim:
+                raise ValueError('Data (shape %s) must have %s dimensions for '
+                                 '%s' % (data.shape, self._data_ndim,
+                                         self.__class__.__name__))
+
+        self._data = data
+        self._tmin = tmin
+        self._tstep = tstep
+        self.vertices = vertices
+        self.verbose = verbose
+        self._kernel = kernel
+        self._sens_data = sens_data
+        self._kernel_removed = False
+        self._times = None
+        self._update_times()
+        self.subject = _check_subject(None, subject, False)
+
+    def __repr__(self):  # noqa: D105
+        s = "%d vertices" % (sum(len(v) for v in self._vertices_list),)
+        if self.subject is not None:
+            s += ", subject : %s" % self.subject
+        s += ", tmin : %s (ms)" % (1e3 * self.tmin)
+        s += ", tmax : %s (ms)" % (1e3 * self.times[-1])
+        s += ", tstep : %s (ms)" % (1e3 * self.tstep)
+        s += ", data shape : %s" % (self.shape,)
+        return "<%s  |  %s>" % (type(self).__name__, s)
+
+    @property
+    def _vertices_list(self):
+        return self.vertices
+
+    @verbose
+    def save(self, fname, ftype='h5', verbose=None):
+        """Save the full SourceTFR to an HDF5 file.
+
+        Parameters
+        ----------
+        fname : string
+            The file name to write the SourceTFR to, should end in
+            '-stfr.h5'.
+        ftype : string
+            File format to use. Currently, the only allowed values is "h5".
+        %(verbose_meth)s
+        """
+        if ftype != 'h5':
+            raise ValueError('%s objects can only be written as HDF5 files.'
+                             % (self.__class__.__name__,))
+        if not fname.endswith('.h5'):
+            fname += '-stfr.h5'
```

Comment:
avoid `if`s without `else`s as much as possible. See this [awesome talk](https://www.youtube.com/watch?v=59YClXmkCVM&t=128s) by @ Katrina Owen

in a case like this one, I would do something like:
```py
fname = fname if fname.endswith('h5') else '{}-stfr.h5'.format(fname)
```
or any way you like to build the fname. (I like this one because it has no spaces)